### PR TITLE
refactor(metrics): rename option property from raiseOnEmptyMetrics to throwOnEmptyMetrics

### PR DESCRIPTION
**Pull request by kozub**
### How to verify this change

Make sure that all existance of "raiseOnEmptyMetrics" were renamed. 

### Related issues, RFCs

[#348](https://github.com/awslabs/aws-lambda-powertools-typescript/issues/348)

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist

- [x] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [x] I have made corresponding changes to the *documentation*
- [x] My changes generate *no new warnings*
- [x] The *code coverage* hasn't decreased
- [ ] I have *added tests* that prove my change is effective and works
- [x] New and existing *unit tests pass* locally and in Github Actions
- [x] Any *dependent changes have been merged and published* in downstream module
- [x] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by dreamorosi** [comment]
Thanks a lot for your first contribution @kozub!

File: packages/metrics/src/Metrics.ts
Comment (dreamorosi) [inline packages/metrics/src/Metrics.ts:229]: Minor: could you please update also the occurrence at [L205](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/996dba61795b526cbfc0b209119b781b80e3c488/packages/metrics/src/Metrics.ts#L205), in the function comment? 

> A decorator automating coldstart capture, **throw** on empty metrics and publishing metrics on handler exit.

File: packages/metrics/src/Metrics.ts
Comment (dreamorosi) [inline packages/metrics/src/Metrics.ts:49]: Minor: [L38](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/996dba61795b526cbfc0b209119b781b80e3c488/packages/metrics/src/Metrics.ts#L38) of the comment has another occurrence of the word `raise`, could you please update that to `throw` too?

File: packages/metrics/examples/empty-metrics.ts
Comment (dreamorosi) [inline packages/metrics/examples/empty-metrics.ts:17]:
```suggestion
  // Since the throwOnEmptyMetrics parameter is set to true, the Powertool throws an Error
```

File: packages/metrics/examples/decorator/empty-metrics.ts
Comment (dreamorosi) [inline packages/metrics/examples/decorator/empty-metrics.ts:22]:
```suggestion
    // Since the throwOnEmptyMetrics parameter is set to true, the Powertool throws an Error
```

File: docs/core/metrics.md
Comment (dreamorosi) [inline docs/core/metrics.md:433]: Minor: There are two more occurrences of the word `raised` at [L277](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/996dba61795b526cbfc0b209119b781b80e3c488/docs/core/metrics.md?plain=1#L277) & [L282](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/996dba61795b526cbfc0b209119b781b80e3c488/docs/core/metrics.md?plain=1#L277), would you mind updating them to `thrown`?

**Review by dreamorosi — CHANGES_REQUESTED** [review]
Left some minor comments & suggestions but looking forward to approve this!

**Comment by kozub** [comment]
Hey @dreamorosi! Thanks for your suggestions. I've added second commit with your comments applied. Please let me know if you prefer to have all that changes as single commit - I can squash it before mering. I'm not sure what's your preference for this repo.

**Review by dreamorosi — DISMISSED** [review]
Looks good to me, thanks for applying the changes. Once we have a second review from another maintainer we'll squash on merge!

**Review by ijemmy — CHANGES_REQUESTED** [review]
Thanks a lot! This issue has been bugging me but we were too busy with the beta release. 

I find 3 more lines. Then we are good to go. (Sorry that examples don't follow the same pattern, we are working on this). 

```
grep -rn -i "raise" . --exclude=\*venv\* --exclude=\*.git\* --exclude=\*node_modules\* --exclude=\*cdk.out\* --exclude=\*.js\* --exclude=\*.d.ts\* --exclude=\*.html\*

./examples/cdk/lib/example-function.MyFunction.ts:32:  metrics.raiseOnEmptyMetrics();
./examples/cdk/lib/example-function.MyFunction.ts:41:  metrics.raiseOnEmptyMetrics();
./examples/cdk/lib/example-function.MyFunctionWithDecorator.ts:18:    raiseOnEmptyMetrics: true,
```

**Comment by kozub** [comment]
Hey @ijemmy  - I've changes method signatures that you mentioned about. Thanks for finding it! I'd tried to search in the project for 'raise' before using VSCode with no results. Looks that I should not trust it ;-)

**Review by dreamorosi — DISMISSED** [review]

**Review by ijemmy — APPROVED** [review]
@kozub  The same here. VSCode seems to miss these. 

I had this issue before and never trusted its search function since then.

**Review by ijemmy — APPROVED** [review]
There was a merge conflict. I have merged from `main`, resolved, and approved.

@dreamorosi  Can I have a new approval?

**Review by dreamorosi — APPROVED** [review]

**Pull request merged**
